### ISSUE TITLE
sourceTable: unsupported list item type: *ast.JoinExpr

### ISSUE THREAD

**Issue by carpenter-ryan** [sqlc-dev/sqlc#1100]
`UPDATE group_calc_totals gct
SET npn = nem.npn
FROM producer_group_attribute ga
         JOIN npn_external_map nem ON ga.npn_external_map_id = nem.id
WHERE gct.group_id = ga.group_id; `

This produces the error, sourceTable: unsupported list item type: *ast.JoinExpr.

The SQL is valid Postgres and executes as expected. I'm using version 1.8 of sqlc.

**Comment by kyleconroy** [comment]
```sql
CREATE TABLE group_calc_totals (
  npn  text,
  group_id text
);

CREATE TABLE producer_group_attribute (
  npn_external_map_id text,
  group_id text
);

CREATE TABLE npn_external_map (
  id   text
);

-- name: Percentile :exec
UPDATE group_calc_totals gct 
SET npn = nem.npn 
FROM producer_group_attribute ga 
JOIN npn_external_map nem ON ga.npn_external_map_id = nem.id 
WHERE gct.group_id = ga.group_id;
```

https://play.sqlc.dev/p/125fcccc7ba6707e40dbabc2d698313f7ec6ea9695672d49e37bd75ffb3352e2

**Comment by DarkHeros09** [comment]
any update on this issue?

**Comment by akhmadnurmuhammad** [comment]
any update on this issue?

**Comment by ahobsonsayers** [comment]
Hate to add to unnecessary notifications, but is there any update on this?

**Comment by ghandic** [comment]
Any news?

**Comment by BuilderHarrison** [comment]
Hello, any update on this?

**Comment by kyleconroy** [comment]
No update yet, but adding this to my list to look for 1.23

**Comment by kyleconroy** [comment]
Fixed in https://github.com/sqlc-dev/sqlc/pull/2841 and will be released in v1.23

**Comment by kyleconroy** [comment]
This is fixed in [v1.23.0](https://docs.sqlc.dev/en/latest/reference/changelog.html#v1-23-0) by enabling the [database-backed query analyzer](https://docs.sqlc.dev/en/latest/howto/generate.html#enhanced-analysis-with-managed-databases). We added a [test case](https://github.com/sqlc-dev/sqlc/tree/main/internal/endtoend/testdata/join_update/postgresql/pgx) for this issue so it won’t break in the future.

You can play around with the [working example on the playground](https://play.sqlc.dev/p/9c3e8169179e0e20d6c0f671bde41687a039f4ef7be87f92345adc6559381b91)
